aks/eks CI cannot save debug artifacts

### Bug Description

See this succesful run https://github.com/canonical/bundle-kubeflow/actions/runs/13912715932/job/38930097133

### To Reproduce

Rerun AKS CI 

### Environment

AKS CI 

### Relevant Log Output

```shell
2025-03-18T01:23:48.5058990Z With the provided path, there will be 82 files uploaded
2025-03-18T01:23:48.5063542Z Artifact name is valid!
2025-03-18T01:23:48.5064704Z Root directory input is valid!
2025-03-18T01:23:48.8168351Z ##[error]Failed to CreateArtifact: Received non-retryable error: Failed request: (409) Conflict: an artifact with this name already exists on the workflow run
```

### Additional Context

_No response_

--- a/.github/workflows/deploy-to-aks.yaml
+++ b/.github/workflows/deploy-to-aks.yaml
@@ -137,6 +137,8 @@ jobs:
       - name: Save debug artifacts
         uses: canonical/kubeflow-ci/actions/dump-charm-debug-artifacts@main
         if: always()
+        with:
+          artifact-prefix: ${{ matrix.bundle_version }}
 
       - name: Get juju status
         run: juju status

--- a/.github/workflows/deploy-to-eks.yaml
+++ b/.github/workflows/deploy-to-eks.yaml
@@ -148,6 +148,8 @@ jobs:
       - name: Save debug artifacts
         uses: canonical/kubeflow-ci/actions/dump-charm-debug-artifacts@main
         if: failure() || cancelled()
+        with:
+          artifact-prefix: ${{ matrix.bundle_version }}
 
       # On failure, capture debugging resources
       - name: Get juju status

